interpreter/src/main/scala/pl/touk/nussknacker/engine/types/EspTypeUtils.scala

```diff
@@ -47,19 +54,19 @@ object EspTypeUtils {
       if (staticMethodsAndFields) publicMethods.filter(membersPredicate.shouldBeVisible).filter(m => Modifier.isStatic(m.getModifiers))
       else publicMethods.filter(membersPredicate.shouldBeVisible).filter(m => !Modifier.isStatic(m.getModifiers))
 
-    val methodNameAndInfoList = filteredMethods.flatMap { method =>
-      val extractedMethod = extractMethod(method)
-      collectMethodNames(method).map(_ -> extractedMethod)
-    }
+    val methodNameAndInfoList = filteredMethods
+      .flatMap(extractMethod(_))
+      .map(method => method.name -> method)
```

Comment:
I don't think method.name is really needed? Maybe let's remove it and leave it as it was before?

Reply:
We need it to generate error messages, because now `MethodInfo.calculateResultType` is responsible for handling errors that were handled elsewere before. I could change `MethodInfo.calculateResultType` to require method's name as argument, and then it could be removed from MethodInfo, but it would look somewhat weird.